### PR TITLE
sanitize file name in object service

### PR DESCRIPTION
**Pull request by sgao640**
@livefire2015

### CODE REVIEW

**Review by livefire2015 — APPROVED** [review]
Summary

  The PR adds filename sanitization to handle international characters and special characters in object storage, plus some example
   updates.

  Code Review

  Positive aspects:

  - Good safety measure: Sanitizing filenames prevents issues with storage systems that don't handle Unicode well
  - Comprehensive character handling: Covers ASCII preservation, Latin character mapping, and fallback to hyphens
  - Appropriate placement: Added to the object service where download URLs are generated

  Concerns:

  1. Potential data loss: Converting non-ASCII characters to ASCII could cause filename collisions or loss of important
  information
  2. No configuration: The sanitization is always applied - users can't opt out or configure behavior
  3. Limited character mapping: Only handles basic Latin characters, other languages get converted to hyphens
  4. No filename uniqueness check: After sanitization, multiple files could end up with the same name

  Suggestions:

  - Consider making sanitization optional or configurable
  - Add collision detection/resolution (e.g., append numbers)
  - Preserve original filename in metadata while using sanitized version for storage
  - Add tests for edge cases (empty strings, very long names, all non-ASCII)

  Minor issues:

  - The example updates seem unrelated to the main change
  - Could benefit from more comprehensive unit tests for the sanitization function

  Overall, this is a reasonable approach to filename handling, but consider the trade-offs between safety and data preservation.

File: pkg/service/object_service.go [hunk 1]
```diff
@@ -232,6 +233,70 @@ func (s *ObjectService) GetUploadURL(ctx context.Context, id uuid.UUID) (string,
 	return backend.GetUploadURL(ctx, object.ObjectKey)
 }
 
+// sanitizeFilename converts a filename to ASCII, replacing non-ASCII characters
+// with their closest ASCII equivalents or removing them if no equivalent exists
+func sanitizeFilename(filename string) string {
```
Comment (livefire2015) [inline pkg/service/object_service.go:238]: Can we move this to utils.go and a public function? @sgao640

File: pkg/service/object_service.go [hunk 2]
```diff
@@ -366,6 +431,25 @@ func (s *ObjectService) GetObjectMetaFromStorage(ctx context.Context, objectID u
 	return objectMeta, nil
 }
 
+// GetObjectMetaFromStorageByObjectKeyAndStorageBackendName retrieves metadata for an object directly from the storage backend
+// This is useful when the upload happens client-side and we need to update our metadata
+func (s *ObjectService) GetObjectMetaFromStorageByObjectKeyAndStorageBackendName(ctx context.Context, objectKey string, storageBackendName string) (*storage.ObjectMeta, error) {
```
Comment (livefire2015) [inline pkg/service/object_service.go:436]: Didn't we already add this in https://github.com/tendant/simple-content/pull/22/files ? @sgao640

Comment (livefire2015) [reply pkg/service/object_service.go:372]: Could you clean it up in this PR? Ping me when it is ready for review again.
